typer/main.py

```diff
@@ -794,6 +833,13 @@ def lenient_issubclass(
     return isinstance(cls, type) and issubclass(cls, class_or_tuple)
 
 
+def is_complex_subtype(type_: Any) -> bool:
+    # For pydantic types, such as `AnyUrl`, there's an extra `Annotated` layer that we don't need to treat as complex
+    return getattr(type_, "__origin__", None) is not None and not is_pydantic_type(
+        type_
+    )
```

Comment:
This function returns `False` for the tests on Python 3.7-3.11, but `True` on Python 3.12, making the unit tests for tutorial cases 003 and 004 fail.

Reply:
At the moment I wrote the pull request, I think 3.12 wasn't officially supported, so I'll loop back on it

Reply:
That'd be great, thanks! 🙏 My first quick digging turns up that `is_pydantic_type` is returning `True` on 3.12 where it was returning `False` on earlier Pythons.

Reply:
OK, it looks like in Python 3.12, the result of Annotated[Type].__module__ is no longer equivalent to `Type.__module`  and I had to add a check in `is_pydantic_type` in case of Annotated